### PR TITLE
Add HTML report generator

### PR DESCRIPTION
**Pull request by marcphilipp**
This PR introduces a new CLI command and API for generating an HTML report out of one or multiple event-based or hierarchical XML reports.

The `Contributor` SPI allows extending the HTML report by third-party libraries which may contribute arbitrary sections for each execution or test node element. For example, JUnit will make use of that as a next step to include its metadata that are defined in an extension schema.

The output is a single-page Vue.js app (using Vite, implemented in Typescript) as a single self-contained HTML file that can be opened in any modern browser:

image

image

A Playwright-based smoke test checks the output for a new sample-test project that uses JUnit's open-test-reporting output.

### CODE REVIEW

File: .github/actions/run-gradle/action.yml [hunk 1]
```diff
@@ -13,13 +13,19 @@ runs:
       with:
         distribution: temurin
         java-version: 17
+    - uses: actions/setup-node@v4
```
Comment (sormuras) [inline .github/actions/run-gradle/action.yml:16]: Will be replaced by renovate with commit hash tag, right?

File: buildSrc/src/main/kotlin/java-conventions.gradle.kts [hunk 2]
```diff
@@ -57,7 +57,7 @@ tasks {
     }
     jar {
         manifest {
-            attributes("Automatic-Module-Name" to "org.opentest4j.reporting.${project.name}")
+            attributes("Automatic-Module-Name" to "org.opentest4j.reporting.${project.name.replace('-', '.')}")
```
Comment (sormuras) [inline buildSrc/src/main/kotlin/java-conventions.gradle.kts:60]: Mh, did we ship `otr` with an illegal automatic module name? Good catch!

Comment (sormuras) [reply buildSrc/src/main/kotlin/java-conventions.gradle.kts:60]: Second thought: maybe `org.opentest4j.reporting` would be better? 🤔

**Review by sormuras — APPROVED** [review]

File: .github/actions/run-gradle/action.yml (hunk 1 above)
Comment (marcphilipp) [reply .github/actions/run-gradle/action.yml:16]: Yep, I was too lazy to do it by hand 😇

File: buildSrc/src/main/kotlin/java-conventions.gradle.kts (hunk 2 above)
Comment (marcphilipp) [reply buildSrc/src/main/kotlin/java-conventions.gradle.kts:60]: No, we didn't have any illegal names before. I've renamed `tooling` to `tooling-core` in this PR so this became necessary.

File: tooling-spi/src/main/java/org/opentest4j/reporting/tooling/spi/htmlreport/Block.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * @param <T> the type of content
+ */
+@JsonTypeInfo(use = SIMPLE_NAME, property = "type")
+public interface Block<T> {
+
+	/**
+	 * {@return the content of this block}
+	 */
+	T getContent();
```
Comment (mpkorstanje) [inline tooling-spi/src/main/java/org/opentest4j/reporting/tooling/spi/htmlreport/Block.java:34]: What kinda of markup is supported here? Only plain text? Markdown? Arbitrary html?(probably a bad idea).

**Pull request merged**